source/extensions/transport_sockets/tls/utility.cc

```diff
@@ -71,13 +71,60 @@ Envoy::Ssl::CertificateDetailsPtr Utility::certificateDetails(X509* cert, const
   return certificate_details;
 }
 
+bool Utility::labelWildcardMatch(absl::string_view dns_label, absl::string_view pattern) {
+  if (pattern.empty()) {
+    return dns_label.empty();
+  }
+
+  // Only valid if wildcard character appear once
+  if (std::count(pattern.begin(), pattern.end(), '*') == 1) {
+    constexpr char glob = '*';
+    // Check the special case of a single * pattern, as it's common.
+    if (pattern.size() == 1 && pattern[0] == glob) {
+      return true;
+    }
+    size_t i = 0, p = 0, i_star = dns_label.size(), p_star = 0;
+    while (i < dns_label.size()) {
+      if (p < pattern.size() &&
+          absl::ascii_tolower(dns_label[i]) == absl::ascii_tolower(pattern[p])) {
```

Comment:
I believe the passed in strings are already lower-case.

Reply:
That is right. I was debating whether to convert (`absl::ascii_tolower`) here or not. But in case this function will be used at some other place then I though let's keep it so that it can be reusable.

Other option I have is to update this function's header comments to indicate that input will already be lower case.

Reply:
removed this `absl::ascii_tolower`.